src/constructs/autoscaling/asg.ts

```diff
@@ -98,7 +99,7 @@ export class GuAutoScalingGroup extends AutoScalingGroup {
       machineImage: { getImage: getImage },
       instanceType: props.instanceType ?? new InstanceType(new GuInstanceTypeParameter(scope, props).valueAsString),
       userData,
-
+      role: new GuInstanceRole(scope, { app: props.app }),
```

Comment:
I think this change will prevent users from attaching a custom role to their ASG? If that's what we want, we should add "role" to the list of [omitted props](https://github.com/guardian/cdk/blob/d896ada6db37661863c0753b0986670492c08a06/src/constructs/autoscaling/asg.ts#L16-L27) to avoid confusing users.

However, I wonder if we should allow users to provide a custom role if they like, and default to `GuInstanceRole` if none is provided?

Reply:
> I wonder if we should allow users to provide a custom role if they like, and default to GuInstanceRole if none is provided?

Jinx! Had the same thought! Have been playing with the `WithDefault` idea too here https://github.com/guardian/cdk/pull/401/files.

Reply:
I thought so too, but having written an additional test to check out this behaviour, it seems like passing in a role manually to `GuAutoScalingGroup` actually overrides the default. I would have assumed that, since `...props` is at the **top** of the object, it would be overridden by the below statements, but for some reason that didn't happen.

Regardless, this is a good point as we want usage to be clear to users. I've just pushed a change that should make it explicitly clear that `role` and `stageDependentProps` are optional and have sensible defaults if not specified.